examples/Spirit/run_spirit_jump.cc

```diff
@@ -261,27 +261,89 @@ void addCost(MultibodyPlant<T>& plant,
   auto x   = trajopt.state();
 
   // Setup the traditional cost function
-  const double R = cost_actuation;  // Cost on input effort
-  const MatrixXd Q = cost_velocity  * MatrixXd::Identity(n_v, n_v); // Cost on velocity
+  trajopt.AddRunningCost( u.transpose()*cost_actuation*u);
 
-  trajopt.AddRunningCost( x.tail(n_v).transpose() * Q * x.tail(n_v) );
-  trajopt.AddRunningCost( u.transpose()*R*u);
+  for(int mode_index = 0; mode_index < trajopt.num_modes(); mode_index++){
+    for(int knot_index = 0; knot_index < trajopt.mode_length(mode_index)-1; knot_index++){
+      drake::solvers::VectorXDecisionVariable xi  = trajopt.state_vars(mode_index, knot_index).tail(n_v);
 
-  for (int joint = 0; joint < 12; joint++){
-    auto power_plus = trajopt.NewSequentialVariable(1, "joint_" + std::to_string(joint)+"_power_plus");
-    auto power_minus = trajopt.NewSequentialVariable(1, "joint_" + std::to_string(joint)+"_power_minus");
+      drake::solvers::VectorXDecisionVariable xip = trajopt.state_vars(mode_index, knot_index+1).tail(n_v);
 
-    trajopt.AddRunningCost(cost_work * (power_plus + power_minus));
+      drake::symbolic::Expression hi = trajopt.timestep(trajopt.get_mode_start(mode_index) + knot_index)[0];
+      drake::symbolic::Expression gi = 0;
+      drake::symbolic::Expression gip = 0;
+      for(int i = 0; i< n_v; i++){
+        gi += cost_velocity * xi[i] * xi[i];
+        gip += cost_velocity * xip[i] * xip[i];
+      }
+
+      trajopt.AddCost(hi/2.0 * (gi + gip));
+    }
+  }
 
-    for(int time_index = 0; time_index < trajopt.N(); time_index++){
-      auto u_i   = trajopt.input(time_index);
-      auto x_i   = trajopt.state(time_index);
+  // Vector of new decision variables
+  std::vector<drake::symbolic::Variable> power_pluses;
+  std::vector<drake::symbolic::Variable> power_minuses;
+
+  // Loop through each joint
+  for (int joint = 0; joint < 12; joint++) {
+    // Loop through each mode
+    for (int mode_index = 0; mode_index < trajopt.num_modes(); mode_index++) {
+      // Create 0th set of power variables
```

Comment:
I see why now tried to think about how I might do like a continue or something to avoid rewriting code 
(if is 0; continue) but didnt parse it carefully enough to know if it worked.

Reply:
I switched some things around, assuming code works will push a simpler fix.